components/server/src/server.rs

```diff
@@ -1493,18 +1566,28 @@ where
                 available = cmp::min(available, disk_stats.available_space());
 
                 let prev_disk_status = disk::get_disk_status(0); //0 no need care about failpoint.
-                let cur_disk_status = if available <= already_full_threshold {
+                let cur_kv_disk_status = if available <= already_full_threshold {
                     disk::DiskUsage::AlreadyFull
                 } else if available <= almost_full_threshold {
                     disk::DiskUsage::AlmostFull
                 } else {
                     disk::DiskUsage::Normal
                 };
+                let cur_disk_status = match (raft_disk_status, cur_kv_disk_status) {
```

Comment:
Move this into a method of `DiskUsage`, e.g. impl std::ops::And<DiskUsage> for DiskUsage

Reply:
Since DiskUsage is generated by proto, just fix it by adding a `calculate_disk_usage` function.